registry/registry.go

```diff
@@ -388,7 +384,14 @@ func (rr *Registry) FindRegistration(c context.Context, key string) (*Registrati
 // Register under the given key. A registration can happen only
 // once, and registering more than once will return an error.
 // Hence, registration can be used for mutual-exclusion.
-func (rr *Registry) Register(c context.Context, key string, options ...Option) error {
+func (rr *Registry) Register(c context.Context, key string, annotations ...map[string]string) error {
+	atns := make(map[string]string)
+	if len(annotations) != 0 || len(annotations) != 1 {
```

Comment:
[NIT] Can simplify this with a `>`:

```suggestion
	if len(annotations) > 1 {
```